database/user.go

```diff
@@ -151,7 +153,36 @@ func (db *DB) UserBySub(ctx context.Context, sub string, create bool) (*User, er
 	users, err := db.managedUsersByField(ctx, "sub", sub)
 	if create && errors.Contains(err, ErrUserNotFound) {
 		var u *User
-		u, err = db.UserCreate(ctx, sub, TierFree)
+		_, email, err := jwt.UserDetailsFromJWT(ctx)
+		if err != nil {
+			// Log the error but don't do anything differently.
```

Comment:
If `err != nil` will the `email` be blank and will that cause an issue with `UserCreate`? 

Is there a test that hits this case?

Reply:
TL;DR: No test. It's not a problem. I have a plan. :)

No, there's no test that hits this use case. This entire project could use some more love on the testing side.

If `err!=nil` the `email` will be empty but that won't cause any issues with `UserCreate` outside of being unable to send a confirmation email and the user will be left unable to reset their password. This was implemented to work with Kratos, which took care of all of those aspects. That's why the identity of the user (i.e. their email) was directly taken from the JWT where Kratos put it.

Now that we're removing Kratos we'll need to change our flows a little and have nginx explicitly call our `POST /user` endpoint instead of creating users implicitly by providing their JWTs. That won't be much of a change for `skynet-webportal` because that call is already being made, it's just being made to Kratos.